--- vaedbl.orig.py	2020-07-31 16:46:14.411382104 +0000
+++ vaedbl.py	2020-07-31 16:46:46.545438670 +0000
@@ -32,7 +32,7 @@
         #  If DB last updated longer than 5 minutes
 
         srcdb = tinydb_src.table('src')
-        tinydb_src.purge_table('src')
+        tinydb_src.drop_table('src')
 
         '''
         Retrieve src hosts
@@ -69,7 +69,7 @@
     if update_needed(os.path.abspath(dest_database), 5):
         #  If DB last updated longer than 5 minutes
         destdb = tinydb_dest.table('dest')
-        tinydb_dest.purge_table('dest')
+        tinydb_dest.drop_table('dest')
 
         '''
         Retrieve detections
